dotcom-rendering/src/lib/article-count.ts

```diff
@@ -5,29 +5,50 @@ import {
 } from '@guardian/support-dotcom-components';
 import { WeeklyArticleHistory } from '@guardian/support-dotcom-components/dist/dotcom/src/types';
 import { hasOptedOutOfArticleCount } from '../web/lib/contributions';
-import { incrementDailyArticleCount } from '../web/lib/dailyArticleCount';
+import {
+	DailyArticleHistory,
+	getDailyArticleCount,
+	incrementDailyArticleCount,
+} from '../web/lib/dailyArticleCount';
+
+export interface ArticleCounts {
+	weeklyArticleHistory: WeeklyArticleHistory | undefined;
+	dailyArticleHistory: DailyArticleHistory | undefined;
+}
 
 // We should monitor this function call to ensure it only happens within an
 // article pages when other pages are supported by DCR.
-export const getArticleCount = async (
+export const getArticleCounts = async (
 	pageId: string,
 	keywordIds: string,
-): Promise<WeeklyArticleHistory | undefined> => {
-	if (await hasOptedOutOfArticleCount()) return;
+): Promise<ArticleCounts> => {
+	if (await hasOptedOutOfArticleCount())
+		return {
+			weeklyArticleHistory: undefined,
+			dailyArticleHistory: undefined,
+		};
 
 	// hasOptedOut needs to be done before we check if articleCount is set in the window
-	// This is because a potential race condition where one invocation of getArticleCount
+	// This is because a potential race condition where one invocation of getArticleCounts
 	// is waiting for hasOptedOut another invocation might receive it and increment the article count.
-	if (!window.guardian.articleCount) {
-		incrementDailyArticleCount();
+	if (!window.guardian.weeklyArticleCount) {
 		incrementWeeklyArticleCount(
 			storage.local,
 			pageId,
 			keywordIds.split(','),
 		);
 
-		window.guardian.articleCount = getWeeklyArticleHistory(storage.local);
+		window.guardian.weeklyArticleCount = getWeeklyArticleHistory(
+			storage.local,
+		);
+	}
+	if (!window.guardian.dailyArticleCount) {
+		incrementDailyArticleCount();
+		window.guardian.dailyArticleCount = getDailyArticleCount();
 	}
 
-	return window.guardian.articleCount;
+	return {
+		weeklyArticleHistory: window.guardian.weeklyArticleCount,
+		dailyArticleHistory: window.guardian.dailyArticleCount,
+	};
```

Comment:
Why couple these two and not just use `getWeeklyArticleCount` and `getDailyArticleCount` separately as two different pieces of state?

A bit of my confusion is that I would expect, consent isn't given to return `undefined`, rather than `EmptyArticleCounts`. Decoupling them might help with confusion.

I can see benefit is we used them together, but we don't seem to anywhere in the code, rather we just destructure this type.

Reply:
Good question, I spent a while thinking about this!
The banner does use both counts, and so it's quite convenient to pass them through together (from `StickBottomBanner`, down a few function calls) here: https://github.com/guardian/dotcom-rendering/pull/3954/files#diff-9fb4db8893a570211b30acefe9224ff36a020986088db2ae69a32dc36723c6e7R235

Also it also felt natural to keep the initialisation of these 2 counts together in this one function because it all depends on awaiting `hasOptedOutOfArticleCount`.
But - I'm not wedded to this implementation if you think it's preferable to split them?

Reply:
Sorry, I think I might have conflated things. 

I am confused by this method return type as it stands. If someone has not consented, I would expect to get `undefined`. Instead I get an object with `undefined` prop values, which feels like they mean that they are available, just not set. So, in my head the return type of `ArticleCounts | undefined` might make more sense.

Some tests might make this more understandable.

On your point re: async `hasOptedOut` -> this is true, dotcom could potentially have a better way of sharing this state. Given that that is a potential performance issue, we could just leave it as 1.

Reply:
Ah right I see what you mean, I'll have a go at refactoring to return `ArticleCounts | undefined`

Reply:
I've just pushed this up